universal-react/server/render.js

```diff
@@ -18,7 +18,8 @@ import urlParser from 'url';
 export default async function renderOnServer(ReactComponent, url, props = {}, cspNonce) {
   const serverContext = {};
   const helmetContext = {};
-  const appData = { url, pageData: {} };
+  const appData = { url, pageData: {}, config: props.config };
```

Comment:
@MatisLepik I don't really like in this implementation that the config is duplicated in both `appData` and `pageData`. As configuration must be still passed to client when server-side rendering is disabled then maybe when can remove config from `pageData`? Although currently when config resides in `pageData` then it can be dynamically updated during route changes. Maybe it's fine when the config is constant and cannot be changed?

Reply:
@allantatter I think the original intention was that create-frontend doesn't have a concept of `config` - it's in the template, but as far as create-frontend knows, it's just random data that comes from the server. Therefore we couldn't mention it in this file.

But I agree that if it's causing problems, it could be moved to `appData`. In practice, every project using this probably has a config

Reply:
@MatisLepik Can we then just remove passing config as props to the App because it's available through AppDataContext? Can AppDataContext available within `App.getPageData()`? This is where we usually need `CMS_URL´, which currently comes there as props.